### ISSUE TITLE
Log should list ALL commits from any branch similar to `git log --all`

### ISSUE THREAD

**Issue by jelaniwoods** [firstdraft/web_git#55]

**Comment by raghubetina** [comment]
Is it... could it be... time to

![](https://s3.amazonaws.com/media-p.slid.es/uploads/318017/images/7233443/pasted-from-clipboard.png)

We may have to iterate over the graph ourselves and draw it?

**Comment by jelaniwoods** [comment]
@raghubetina I'm not sure how we'd do that using the gem at least. It doesn't seem possible to get the graph itself. Unless you're referring to building the graph by going to each branch and grabbing all the commits? That sounds... intense 😯

**Comment by raghubetina** [comment]
@jelaniwoods Yes indeed, that is what I was referring to. Traversing the graph ourselves, commit by commit if necessary. This is how git itself does it — each commit knows who its parent(s) is, and git follows that back to draw the log or graph.

I don't know what commands are available in the gem, but I speculate that we could simplify producing the graph with something like:

 - listing all branches
 - For each branch, `git log` and save the list of commit SHAs
 - Find common commit SHAs
 - Merge the logs at those points

Just spitballin' though.

**Comment by raghubetina** [comment]
@jelaniwoods Just checking in on this — any progress? I would like to do a session on how to jump back and forth between commits this week or next.

**Comment by jelaniwoods** [comment]
@raghubetina I made some progress. I have a list of Hashes, where each Hash represents a branch. The Hash has a key for branch name, HEAD, and all the commit SHA's from the branch. I've combined them into one list of all commit SHAs but still need to find the HEADs of all branches and verify the order of the commits.

**Comment by raghubetina** [comment]
@jelaniwoods Okay. If we iterate over each branch and do a `git log`, would that give us the list of commits in the correct order?

**Comment by jelaniwoods** [comment]
@raghubetina  I assumed that would be the case, but either I made an error with the logic somewhere or `--graph` sorts it differently. The order I got was slightly different.

<img width="1051" alt="Screen Shot 2020-05-18 at 5 10 38 PM" src="https://user-images.githubusercontent.com/17581658/82264492-9ecafa00-992a-11ea-851a-8d5068adc6a6.png">

**Comment by raghubetina** [comment]
@jelaniwoods Where do we stand on drawing the whole graph in web_git?

**Comment by jelaniwoods** [comment]
@raghubetina ah! I didn't see this. I had paused this since last quarter. I believe the last issue to figure out is the order to create the branches, since the graphing library needs to create branches at specific times. I'll focus back on this.
